app/src/main/java/com/android/sample/ui/mainscreen/ViewAllScreen.kt

```diff
@@ -211,10 +217,7 @@ fun ViewAllCard(
               .clickable {
                 viewModel.selectWorkout(workout)
 
-                when (workout) {
-                  is BodyWeightWorkout -> navigationActions.navigateTo(Screen.BODY_WEIGHT_WORKOUT)
-                  is YogaWorkout -> navigationActions.navigateTo(Screen.YOGA_WORKOUT)
-                }
+                onClick(workout, viewModel, navigationActions)
```

Comment:
I think it makes more sense to remove `viewModel.selectWorkout(workout)` and make the assumption that it must be done in the `onClick(workout, viewModel, navigationActions)`.

Imagine you want to use this composable to just see the list of workouts, when you will click on a card it will use the viewmodel for select the workout, but it is not expected to do that.